Apps/W1/Shopify/app/src/Companies/Codeunits/ShpfyCompByEmailPhone.Codeunit.al

```diff
@@ -27,6 +29,61 @@ codeunit 30304 "Shpfy Comp. By Email/Phone" implements "Shpfy ICompany Mapping"
         exit('');
     end;
 
+    internal procedure FindMapping(var ShopifyCompany: Record "Shpfy Company"; var TempShopifyCustomer: Record "Shpfy Customer" temporary): Boolean
+    var
+        Customer: Record Customer;
+        ShopifyCustomer: Record "Shpfy Customer";
+        CustomerMapping: Codeunit "Shpfy Customer Mapping";
+        PhoneFilter: Text;
+    begin
+        if not IsNullGuid(ShopifyCompany."Customer SystemId") then
+            if Customer.GetBySystemId(ShopifyCompany."Customer SystemId") then
+                exit(true)
+            else begin
+                Clear(ShopifyCompany."Customer SystemId");
+                ShopifyCompany.Modify();
+            end;
+
+        if IsNullGuid(ShopifyCompany."Customer SystemId") then begin
+            if TempShopifyCustomer.Email <> '' then begin
```

Comment:
These could be extracted to FindByEmail and FindByPhoneNo procedures

Reply:
I know it shows as a insertion, but this code is simply moved from another place.
Should we change the code structure in this case? (Just thinking about code comparison and code review from their side)

Reply:
My opinion is yes. If we would address this in an internal PR, then we can address it for MS as well. Any code we touch, try to make it better.

Reply:
extracted to separate procedures